watcher.close() throw the EtcdException(InterruptedException)

Jetcd will throw an `EtcdException` if you close an on going listened watcher.
The `InterruptedException` exception caused by `this.eventsQueue.take();` when you interrupted the thread.
For `this.eventsQueue.take();` , it is an exception if the thread is interrupted. But for Jetcd the expected behavior for the `watcher.close()` method is to get the thread interrupted. So why does Jetcd raise an `EtcdException` exception?

`Caused by: java.lang.InterruptedException
	at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:404)
	at java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at com.coreos.jetcd.internal.impl.WatchImpl$WatcherImpl.listen(WatchImpl.java:449)
	... 4 more`

--- a/src/main/java/com/coreos/jetcd/internal/impl/WatchImpl.java
+++ b/src/main/java/com/coreos/jetcd/internal/impl/WatchImpl.java
@@ -87,7 +87,9 @@ class WatchImpl implements Watch {
 
   @Override
   public synchronized void close() {
-    Preconditions.checkState(!isClosed(), "Watch client has been closed");
+    if (isClosed()) {
+      return;
+    }
 
     this.setClosed();
     this.notifyWatchers(EtcdExceptionFactory.newEtcdException("Watch client has been closed"));
@@ -419,7 +421,9 @@ class WatchImpl implements Watch {
     @Override
     public void close() {
       synchronized (this.closedLock) {
-        Preconditions.checkState(!isClosed(), "Watcher has been closed");
+        if (isClosed()) {
+          return;
+        }
 
         this.setClosed();
       }
@@ -435,6 +439,12 @@ class WatchImpl implements Watch {
       try {
         return this.createWatchResponseFuture().get();
       } catch (ExecutionException e) {
+        synchronized (this.closedLock) {
+          if (isClosed()) {
+            // returns EtcdException indicates watcher has been closed.
+            throw EtcdExceptionFactory.newEtcdException("Watcher has been closed");
+          }
+        }
         Throwable t = e.getCause();
         if (t instanceof CompactedException) {
           throw (CompactedException) t;

